orchagent/neighorch.cpp

```diff
@@ -246,15 +266,18 @@ bool NeighOrch::addNextHop(const IpAddress &ipAddress, const string &alias)
     next_hop_entry.nh_flags = 0;
     m_syncdNextHops[nexthop] = next_hop_entry;
 
-    m_intfsOrch->increaseRouterIntfsRefCount(alias);
+    m_intfsOrch->increaseRouterIntfsRefCount(nexthop.alias);
 
-    if (ipAddress.isV4())
+    if (!nexthop.isMplsNextHop())
     {
-        gCrmOrch->incCrmResUsedCounter(CrmResourceType::CRM_IPV4_NEXTHOP);
-    }
-    else
-    {
-        gCrmOrch->incCrmResUsedCounter(CrmResourceType::CRM_IPV6_NEXTHOP);
+        if (nexthop.ip_address.isV4())
+        {
+            gCrmOrch->incCrmResUsedCounter(CrmResourceType::CRM_IPV4_NEXTHOP);
```

Comment:
for mpls type nexthop there is no CRM accounting ?

Reply:
@abdosi That is correct.  There is no support for MPLS NextHop CRM accounting.

Reply:
2 further comments:

a) What is the reason of not implementiong for SAI to have MPLS CRM nexthop ? Is there plan ?

b) If we do not have Nexthop CRM Accounting how does Nexthop Group/Nexthop Group Member CRM accounting is handled. I see as of now that it will increment for MPLS also. Is my understanding correct ? not sure is it correct behaviour

Reply:
@abdosi In all the MPLS HLD reviews, no one has requested MPLS MH CRM support.  I do not see any coordination between NHG and NH CRM accounting, so not sure it makes any difference if MPLS NHGMs are counted but there is no MPLS-specific NH CRM.

Reply:
somehow i feel this gives wrong impression from application. i feel if the member of nexthop group is mpls type it also should nit be accounted from CRM perspective. @prsunny what do you think ?

Reply:
@abdosi Please let me know if this is considered a blocking issue or if it can be addressed in a follow-on PR.

Reply:
@abdosi @prsunny CRM support for MPLS NHs is now included.